firmware/common/ui_painter.cpp

```diff
@@ -58,7 +58,7 @@ int Painter::draw_string(
     for (auto c : text) {
         if (escape) {
             if (c <= 15)
```

Comment:
Maybe use like `std::size(term_colors)` so it's clear where this number comes from?

Reply:
Also, look in ui_widget.cpp -- that's where Console handles escape characters.